ihatemoney/web.py

```diff
@@ -647,7 +647,11 @@ def list_bills():
     csrf_form = EmptyForm()
     # set the last selected payer as default choice if exists
     if "last_selected_payer" in session:
-        bill_form.payer.data = session["last_selected_payer"]
+        if g.project.id in session["last_selected_payer"]:
+            bill_form.payer.data = session["last_selected_payer"][g.project.id]
```

Comment:
*Question for Flask experts:*
When does the session expires? I changed the representation of `session["last_selected_payer"]`, does the session automatically expires during an update of IHM?

Reply:
No, the session is kept even when you update IHM, so you always need to account for the old format and convert it. This has been a frequent source of heisenbugs in the past.

In general, can you separate the two changes into separate commits/PR? One which adds the new "payed_for", and one which fixes the behaviour of payer.

Reply:
> In general, can you separate the two changes into separate commits/PR?

I'll do that.

Reply:
I updated the PR. The bug fix is now in #1224 and the new feature in this PR.